interval/interval.go

```diff
@@ -15,9 +15,38 @@
 // Package interval provides a struct representing a time interval.
 package interval
 
-import "time"
+import (
+	"math/rand"
+	"time"
+)
 
 // Interval represents the interval [Start, End).
 type Interval struct {
 	Start, End time.Time
 }
+
+// RandomInstant returns a random time (to second precision) that falls within
+// the Interval.
+//
+// If the Interval is nil, or Interval.End <= Interval.Start, the zero value
+// time.Time will be returned.
+func (i *Interval) RandomInstant() time.Time {
+	if i == nil {
+		return time.Time{}
```

Comment:
This is nice, and almost lets you omit the `if instant != nil` check in the calling code.
Except, you still want to fall back on the timeNow + validityPeriod default in that case.
I can think of ways around that but they make things less readable, so keep as-is.